server/images.go

```diff
@@ -1052,7 +1052,7 @@ func PullModel(ctx context.Context, name string, regOpts *RegistryOptions, fn fu
 	// build deleteMap to prune unused layers
 	deleteMap := make(map[string]bool)
 
-	if noprune = os.Getenv("OLLAMA_NOPRUNE"); noprune == "" {
+	if noprune := os.Getenv("OLLAMA_NOPRUNE"); noprune == "" {
```

Comment:
I believe this is used later on in the function
```suggestion
	if noprune = os.Getenv("OLLAMA_NOPRUNE"); noprune == "" {
```